### DISCUSSION TITLE
自定义工具链编译器不支持 @flags 传入选项文件

### DISCUSSION THREAD

**Discussion by 2531** [xmake-io/xmake#4855] [Q&A] [answered]
我自定义了一个工具链，现在编译没有问题，但是链接不了，看log分析应该编译器不支持使用@添加选项文件，我使用的编译器用 `-f file` 这样的格式传入选项文件，看了一遍教程文档没找到解决方法，所以想问问有没有啥解决办法。

![image](https://github.com/xmake-io/xmake/assets/35393173/432ae3ab-79d2-4f70-9e4e-b528a4cd092e)
![image](https://github.com/xmake-io/xmake/assets/35393173/61a4238a-832c-420c-b6d8-69937b15bc21)

**Comment by waruqi** [comment, marked as answer]
windows 上过长的命令行有限制，目前仅仅对已知的 clang/gcc/cl 做了处理，如果自定义 toolchain 是 gcc 系的，就能支持，如果是完全未知的，目前支持不了，除非提 pr ，让 xmake 内置支持它
